src/contracts/socket/index.ts

```diff
@@ -0,0 +1,166 @@
+  });
+  if (!quote) {
+    throw new Error('No quote found');
+  }
+  console.log('🔗 Socket quote:', quote);
+  // check if routes are found
+  if (!quote.result.routes.length) {
+    throw new Error('No routes found');
+  }
+  // check if only single user tx is present
+  if (quote.result.routes[0].userTxs.length > 1) {
+    throw new Error('Multiple user txs found');
+  }
+  // check if the user tx is fund-movr
+  if (quote.result.routes[0].userTxs[0].userTxType !== 'fund-movr') {
+    throw new Error('User tx is not fund-movr');
+  }
+
+  // use the first route to prepare the bridge tx
+  const route = quote.result.routes[0];
+  const txData = await getBungeeRouteTransactionData(route);
+  const { routeId, encodedFunctionData } = decodeBungeeTxData(
+    txData.result.txData
+  );
+
+  // Create bridged token contract
+  const bridgedTokenContract = WeirollContract.createContract(
+    getErc20Contract(sourceToken),
+    CommandFlags.CALL
+  );
+
+  // Get balance of CoW shed proxy
+  console.log(
+    `[socket] Get cow-shed balance for ERC20.balanceOf(${cowShedAccount}) for ${bridgedTokenContract.address}`
+  );
+
+  // Get bridged amount (balance of the intermediate token at swap time)
+  const sourceAmountIncludingSurplusBytes = planner.add(
+    bridgedTokenContract.balanceOf(cowShedAccount).rawValue()
+  );
+
+  // Check & set allowance for SocketGateway to transfer bridged tokens
+  // check if allowance is sufficient
+  const {
+    approvalData: {
+      approvalTokenAddress,
+      allowanceTarget,
+      minimumApprovalAmount,
+    },
+  } = txData.result;
+  const intermediateTokenContract = getErc20Contract(
+    approvalTokenAddress,
+    await getWallet(sourceChain)
+  );
+  const allowance = await intermediateTokenContract.allowance(
+    cowShedAccount,
+    allowanceTarget
+  );
+  console.log('current cowshed allowance', allowance);
+  if (allowance < minimumApprovalAmount) {
+    // set allowance
+    const approvalTokenContract = WeirollContract.createContract(
+      getErc20Contract(approvalTokenAddress),
+      CommandFlags.CALL
+    );
+    console.log(
+      `[socket] approvalTokenContract.approve(${allowanceTarget}, ${sourceAmountIncludingSurplusBytes}) for ${approvalTokenContract}`
+    );
+    const allowanceToSet = ethers.utils.parseUnits(
+      '1000',
+      await intermediateTokenContract.decimals()
+    );
+    planner.add(approvalTokenContract.approve(allowanceTarget, allowanceToSet));
+  }
+
+  const bytesLibContractAddress = '0x8f6BA63528De7266d8cDfDdec7ACFA8446c62aB4';
+  const bytesLibContract = WeirollContract.createContract(
+    new ethers.Contract(bytesLibContractAddress, bytesLibAbi),
+    CommandFlags.CALL
+  );
+  const encodedFunctionDataWithNewAmount = planner.add(
+    bytesLibContract.replaceBytes(
+      encodedFunctionData,
+      4, // first 4 bytes are the function selector
+      32, // first 32 bytes of the params are the amount
+      sourceAmountIncludingSurplusBytes
```

Comment:
How does this work? 

We update the sell amount, but does it approve the buy amount too so we get more tokens?

Reply:
We're not updating the sell amount here. We check the cowshed balance and use it to update the input amount for the bridge contract function call here.

Reply:
Output amount on the destination chain depends on this bridge input amount

Reply:
> Output amount on the destination chain depends on this bridge input amount

If we just modify the input amount, what guarantees the rate the user receive?

Reply:
If we show the user a quote where they can bridge 100 DAI in ethereum and receive 100 DAI in Arbitrum (1:1).
If we get 110 DAI instead, does the user have any guarantees they get at least 10 extra DAI in Arbutrum?

Reply:
Sample Across swap txns:
- Sample 1
  - https://explorer.cow.fi/orders/0x8a6acb1e6c9944846b7cd4bcf4f0053b64e65fdbbaeb61860125ec770d6c7111daee4d2156de6fe6f7d50ca047136d758f96a6f067ec5757?tab=overview
  - https://www.socketscan.io/tx/0x2e840ae6842b362cfdb3d6d887fba5ff5c3933d59bad72f4f5bb56f249e9c34c

- Sample 2 
  - https://explorer.cow.fi/orders/0x8741efeef0b160139c7bfdde09a7c9013789143b0ba4dbb53b494f6495b944badaee4d2156de6fe6f7d50ca047136d758f96a6f067ec590f?tab=overview
  - https://www.socketscan.io/tx/0x398e7851814e46e2883c8870be84978e6883adef62bbbb8de46a435af0a9bdb9